src/genn/backends/cuda/backend.cc

```diff
@@ -534,6 +537,18 @@ void Backend::genDefinitionsPreambleInternal(CodeStream &os, const ModelSpecMerg
     if(getRuntimeVersion() >= 9000) {
         os <<"#include <cuda_fp16.h>" << std::endl;
     }
+    
+    // If NVTX profiling is enabled, include nvToolsExt header
+    if(getPreferences<Preferences>().enableNVTX) {
+        os << "#include <nvToolsExt.h>" << std::endl;
+        
+        // Add helper macros for NVTX ranges
+        os << std::endl;
+        os << "// NVTX profiling macros" << std::endl;
+        os << "#define NVTX_PUSH(name) nvtxRangePushA(name)" << std::endl;
+        os << "#define NVTX_POP() nvtxRangePop()" << std::endl;
```

Comment:
These macros don't really seem to be necessary - they don't save any typing and code generation is handling turning off profiling i.e. you don't just disable it by:
```c++
#define NVTX_PUSH(name)
#define NVTX_POP()
```